.github/workflows/puppet.yml

```diff
@@ -0,0 +1,47 @@
+name: Unit tests
```

Comment:
it feels weird to have two files, both having `name: Unit tests`, and one `unit` job, but doing different things.

IMHO this should be:
1. one file called `main.yml` or `tests.yml` or whatever
2. having a useful name (or just "CI")
3. having two jobs defined: `puppet` and `jenkins`.